dsp-user/handler/src/test/scala/dsp/user/handler/UserHandlerSpec.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dsp.user.handler
+
+import zio.ZLayer
+import zio._
+import zio.test._
+import dsp.user.domain._
+import dsp.user.repo.impl.UserRepoMock
+import dsp.user.domain.User
+import dsp.valueobjects.User._
+
+/**
+ * This spec is used to test [[dsp.user.handler.UserHandler]].
+ */
+object UserHandlerSpec extends ZIOSpecDefault {
+
+  def spec = (userTests)
+
+  private val givenNameZio  = GivenName.make("GivenName1").toZIO
```

Comment:
why so complicated? here you could create plain value-objects and use them directly in `userHandler.create` and the assertions. Or am I missing something?

Reply:
The problem was that make() returns a Validation. But we agree that it looks complicated. We replaced it now with fold().